vllm/model_executor/models/siglip_so400m.py

```diff
@@ -0,0 +1,590 @@
+            config, quant_config, prefix=f"{prefix}.head")
+
+    def forward(self, pixel_values: torch.Tensor,
+                patch_attention_mask: torch.Tensor) -> torch.Tensor:
+        x = self.embeddings(pixel_values, patch_attention_mask)
+        for layer in self.encoder.layers:
+            x = layer(x, attention_mask=None)
+        x = self.post_layernorm(x)
+        pooled_output = self.head(x, attention_mask=patch_attention_mask)
+        return pooled_output
+
+
+class SiglipTextEmbeddings(nn.Module):
+
+    def __init__(self, config: SiglipTextConfig):
+        super().__init__()
+        self.token_embedding = VocabParallelEmbedding(config.vocab_size,
+                                                      config.hidden_size)
+        self.position_embedding = VocabParallelEmbedding(
+            config.max_position_embeddings, config.hidden_size)
+        self.register_buffer(
+            "position_ids",
+            torch.arange(config.max_position_embeddings).expand((1, -1)),
+            persistent=False)
+
+    def forward(self, input_ids: torch.Tensor) -> torch.Tensor:
+        if input_ids.dim() == 1:
+            input_ids = input_ids.unsqueeze(0)
+        seq_length = input_ids.shape[1]
+        position_ids = self.position_ids[:, :seq_length].to(input_ids.device)
+        token_embeds = self.token_embedding(input_ids)
+        position_embeds = self.position_embedding(position_ids)
+        embeddings = token_embeds + position_embeds
+        return embeddings
+
+
+class SiglipEncoder(nn.Module):
+
+    def __init__(self, config, quant_config, prefix):
+        super().__init__()
+        self.layers = nn.ModuleList([
+            SiglipEncoderLayer(config,
+                               quant_config,
+                               prefix=f"{prefix}.layers.{i}")
+            for i in range(config.num_hidden_layers)
+        ])
+
+
+class SiglipTextTower(nn.Module):
+
+    def __init__(self,
+                 config: SiglipTextConfig,
+                 quant_config: Optional[QuantizationConfig] = None,
+                 prefix: str = "text_model"):
+        super().__init__()
+        self.config = config
+        self.embeddings = SiglipTextEmbeddings(config)
+        self.encoder = SiglipEncoder(config,
+                                     quant_config,
+                                     prefix=f"{prefix}.encoder")
+        self.final_layer_norm = nn.LayerNorm(config.hidden_size,
+                                             eps=config.layer_norm_eps)
+        projection_size = getattr(config, 'projection_size',
+                                  config.hidden_size)
+        self.head = ColumnParallelLinear(config.hidden_size,
+                                         projection_size,
+                                         bias=True,
+                                         quant_config=quant_config,
+                                         prefix=f"{prefix}.head")
```

Comment:
```suggestion
        self.head = ColumnParallelLinear(config.hidden_size,
                                         projection_size,
                                         bias=True,
                                         gather_output=True,
                                         quant_config=quant_config,
                                         prefix=f"{prefix}.head")
```
Otherwise TP won't work.